platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/platforms/PlatformResolveIntegrationTest.groovy

```diff
@@ -717,6 +717,26 @@ class PlatformResolveIntegrationTest extends AbstractHttpDependencyResolutionTes
         //Shape of the graph is not checked as bug was failing resolution altogether
     }
 
+    def foo() {
```

Comment:
Something about these dependencies would cause the engine to absolutely blow up. This dependency graph would cause reentrant selection behavior. Now this no longer happens but we should probably minimize this